src/components/Modals/ModalBase.jsx

```diff
@@ -0,0 +1,46 @@
+// React Imports
+import React from 'react';
+// Material UI Imports
+import Dialog from '@mui/material/Dialog';
+import Drawer from '@mui/material/Drawer';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useTheme } from '@mui/material/styles';
+
+/**
+ * ModalBase - Component that renders either a modal dialog or drawer based on screen size
+ *
+ * @memberof Modals
+ * @name ModalBase
+ * @param {object} Props - Props for ModalBase component
+ * @param {boolean} Props.open - Boolean for opening the modal
+ * @param {string} Props.arialabelledby - ID of the element that labels the modal
+ * @param {string} Props.ariadescribedby - ID of the element that describes the modal
+ * @param {Function} Props.onClose - Function to call when the modal is closed
+ * @param {React.JSX.Element} Props.children - Content to be displayed within the modal
+ * @returns {React.JSX.Element} The rendered modal component
+ */
+const ModalBase = ({ open, arialabelledby, ariadescribedby, onClose, children }) => {
+  const theme = useTheme();
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
+  const DialogOrDrawer = isSmallScreen ? Drawer : Dialog;
```

Comment:
I was curious how you manage to get the condition media query condition set, this is neat. 🥳

Reply:
To be honest someone else set it up and I'm just using it. I think it was Scott.